### ISSUE TITLE
Cilium does not ignore namespace labels even if they are set with labels option

### ISSUE THREAD

**Issue by aanm** [cilium/cilium#7008]
example:
```
    args:
      --debug=$(CILIUM_DEBUG)
      --kvstore=etcd
      --kvstore-opt=etcd.config=/var/lib/etcd-config/etcd.config
      --labels=!io.cilium.k8s.namespace.labels.field.cattle.io/projectId=projid
```

**Comment by aanm** [comment]
I was able to use the filter `--labels` correctly, setting the kube-system with 2 labels, `foo=bar` and `hello=world`

```
$ kubectl get ns kube-system --show-labels
NAME          STATUS   AGE    LABELS
kube-system   Active   2d3h   foo=bar,hello=world
```

then I've started cilium with `--labels=!io.cilium.k8s.namespace.labels.foo` to ignore all namespace labels that start with `foo` which results in the following endpoint:
```
3513       Disabled           Disabled          54415      k8s:io.cilium.k8s.namespace.labels.hello=world    f00d::a10:0:0:27b8   10.16.118.102   ready   
                                                           k8s:io.cilium.k8s.policy.cluster=default                                                       
                                                           k8s:io.cilium.k8s.policy.serviceaccount=coredns                                                
                                                           k8s:io.kubernetes.pod.namespace=kube-system                                                    
                                                           k8s:k8s-app=kube-dns                                                                           
```

After restarting cilium-agent with `--labels=!io.cilium.k8s.namespace.labels`, to ignore all namespace labels, resulted in the following endpoint:

```
3513       Disabled           Disabled          104        k8s:io.cilium.k8s.policy.cluster=default          f00d::a10:0:0:27b8   10.16.118.102   ready   
                                                           k8s:io.cilium.k8s.policy.serviceaccount=coredns                                                
                                                           k8s:io.kubernetes.pod.namespace=kube-system                                                    
                                                           k8s:k8s-app=kube-dns
```

notice the endpoint is the same except its security ID which was derived from the set of labels except the namespace labels.

I would suggest to use the filter `--labels=!io.cilium.k8s.namespace.labels.field.cattle.io/projectId` instead of `--labels=!io.cilium.k8s.namespace.labels.field.cattle.io/projectId=projid`.

**Comment by asauber** [comment]
@aanm Thanks! I'll try this.

**Comment by tgraf** [comment]
We should try and detect this automatically and make it the default. It's impossible for users to know this in advance.

**Comment by mechpen** [comment]
We were trying to bootstrap cilium in kubernetes clusters with the `cilium-etcd-operator`.  Our `kube-system` namespace has a few labels that causes the bootstrap failure: the `kube-dns`, `etcd-operator` and `cilium-etcd` pods could not come up.

The problem is that when assigning well-known identities to pods, cilium uses `lbls.Equal()` function to match pod labels with the identity labels.  This function returns false for the bootstrap pods, becauses the pods have extra labels from namespace labels (with prefix `k8s:io.cilium.k8s.namespace.labels`). Thus no identity could be assigned to the pods.

I would expect that the labels are used as selectors. If a pod labels contain all the identity labels, the pod could assume this identity.  It shouldn't matter if the pod has more labels.

**Comment by tgraf** [comment]
@mechpen The problem is that if we ignore certain labels then any pod assigned a well-known identity would ignore any policy you specify that matches on the namespace label. We are careful to do that but we may to fall back to this option.

**Comment by taharah** [comment]
@tgraf I ran into the same issue described by @mechpen. It took me awhile to debug why the pods were failing to start when just trying to follow the Getting Started guide. Setting the ignore labels flag is a good workaround for now. I'd like to see the labels for the well-known identities be less restrictive or maybe even configurable going forward. In the interim at least a note in the documentation as something to be aware of would have been immensely helpful.

**Comment by ianvernon** [comment]
Removing backport labels for the following release branches / reasons:

* v1.4 is moving to a security-fix-only backport model

I am going to nominate this for v1.6 backport so we can track this to see if it still is a problem. I will not nominate this for v1.5 backport because I'd like to reduce the churn in release branches before the most recent release (e.g., 1.6 is the most recent branch, do not backport such fixes to v1.5).

**Comment by joestringer** [comment]
Closing due to age. Post a comment to request re-opening the investigation of this issue.
